### ISSUE TITLE
Adjusting thr_r2, max_size, max_K during snp_ldsplit( ) calculation

### ISSUE THREAD

**Issue by Sohambiswas8** [privefl/paper-ldsplit#1]
Hi Florian, 

Thank you so much for the bigsnpr package. However I am facing issues in calculating correlation matrix and splitting an entire chromosome into independent LD blocks. 

I have in-house SNP genotype data for Indian populations. I want to test 'snp_ldsplit' only for chromosome 22. In my data, CHR22 contains only 8317 variants for 475 individuals. I have calculated 'snp_cor', with thr_r2 as default (0). The result corr matrix looks like this:
`corr <- snp_cor(G, ind.row = rows_along(G), ind.col = ind.snp, infos.pos = genedist_mod[ind.snp]`
```
> corr0[1:6,1:6]
6 x 6 sparse Matrix of class "dsCMatrix"
[1,]  1.00000000  0.273100323 -0.034557137 -0.0621476 -0.05520707 -0.07121888

[2,]  0.27310032  1.000000000  0.005258147 -0.0892072 -0.08623673 -0.07075284

[3,] -0.03455714  0.005258147  1.000000000 -0.2988284 -0.30534625 -0.23395518

[4,] -0.06214760 -0.089207198 -0.298828410  1.0000000  0.99224619 -0.28822473

[5,] -0.05520707 -0.086236732 -0.305346247  0.9922462  1.00000000 -0.29486812

[6,] -0.07121888 -0.070752840 -0.233955182 -0.2882247 -0.29486812  1.00000000

```

I have converted the genomic positions (GRCh37) from bp to cM in the map file. But when I am calculating snp_ldsplit, it gives me NULL result. I have set thr_r2 as default previously, max_size = 100 and max_K = 100 (as I have 8317 variants). Can you help me figure out where I am making a mistake?
(If it helps, I am writing the entire code block I have tried so far.....)
```
library(bigsnpr) 
file <- snp_readBed("postqcCHR22.bed")
obj.bigSNP <- snp_attach("postqcCHR22.rds")
G   <- obj.bigSNP$genotypes
CHR <- obj.bigSNP$map$chromosome
POS <- obj.bigSNP$map$physical.pos
genedist <- obj.bigSNP$map$genetic.dist
genedist_mod <- snp_asGeneticPos(CHR, POS)
ind.snp <- cols_along(G)
corr <- snp_cor(G, ind.row = rows_along(G), ind.col = ind.snp, infos.pos = genedist_mod[ind.snp])
corr[1:6,1:6]
MAX_SIZE <- 100
MAX_BLOCK <- 100
block_split <- bigsnpr::snp_ldsplit(corr, thr_r2 = 0, min_size = 1, max_size = MAX_SIZE, max_K = MAX_BLOCK)

```

Regards,
Soham

**Comment by privefl** [comment]
- If you use the genetic positions in cM, you should probably also set the size to e.g. 3/1000 in `snp_cor()`. You'll get a sparser matrix, but that's not the issue here.
- Usually you can set `thr_r2` to e.g. 0.02 because you do not really care about small correlations when splitting in blocks. But again, that's not the problem here.
- The problem is that there is no solution that meet the conditions on `min_size`, `max_size`, `max_r2`, etc. I guess I would try larger values for `max_size` here; you should be able to provide a vector directly to try many different values.

**Comment by Sohambiswas8** [comment]
Hi, 

Thank you for the updates and I am sorry for the late reply. After some trials with `min_size`, `max_size`, ' max_size' and `max_r2` values, I have finally defined an LD boundary. Just one thing I need to know. Before computing correlation between SNPs I used snp_asGeneticPos( ) to obtain the interpolated genetic map from my PLINK map file. Does this function also support GRCh37 build or is it explicitly for GRCh38?
